apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile.
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	// +optional
+	DisplayName string `json:"displayName"`
+
+	// ClusterManager defines which cluster manager owns this ClusterProfile resource
+	// +required
```

Comment:
We had multiple lengthy SIG-MC meeting discussions on how to avoid Cluster Managers from stepping on each other and the current solution is to make sure we know which Cluster Manager owns a ClusterProfile. WDYT?

Reply:
I guess the controller would have to set it by default to none or to the provider of the controller. but ok, I agree on having it `required`